anki-editor.el

```diff
@@ -243,6 +247,96 @@ Raise an error if applicable."
     (when .error (error .error))
     .result))
 
+(defvar anki-editor--api-active-queue 1
+  "Determines which anki-editor--api-request-queue is accepting requests.")
+(defvar anki-editor--api-request-queue-1 nil
+  "Queued requests for anki-editor-api-dispatch-queue.")
+(defvar anki-editor--api-request-queue-2 nil
+  "Queued requests for anki-editor-api-dispatch-queue.")
+
+(defun anki-editor-api--make-queued-request (request success error)
+  (list :request request :success success :error error))
+
+(defun anki-editor-api--get-active-queue ()
+  (if (= anki-editor--api-active-queue 1)
+      anki-editor--api-request-queue-1
+    anki-editor--api-request-queue-2))
+
+(defun anki-editor-api--push-active-queue (request)
+  (if (= anki-editor--api-active-queue 1)
+      (push request anki-editor--api-request-queue-1)
+    (push request anki-editor--api-request-queue-2)))
+
+(defun anki-editor-api--toggle-active-queue ()
+  (if (= anki-editor--api-active-queue 1)
+      (setq anki-editor--api-active-queue 2
+            anki-editor--api-request-queue-1 nil)
+    (setq anki-editor--api-active-queue 1
+          anki-editor--api-request-queue-2 nil)))
+
+(cl-defun anki-editor-api-enqueue-request (action params &rest callbacks &key success error)
```

Comment:
We seem to have two ways to batch and queue up API calls now (the other is `anki-editor-api-enqueue`); this seems kind of suboptimal.

Reply:
Yeah, it's not great. I wrote a bit about the limitations of the old enqueue/dispatch system and the trade-offs for doing it this new way in the commit message here: 8fb89db545373c342f2a2b4ec2b008c86ad1d7a3. I'm planning to rip out the old system when I get some more time but it's tied up in a couple other macros (mainly --with-collection-data-updated) that I haven't fully wrapped my head around just yet. I thought the performance improvements to -push-notes were at least worth talking about before then though.

Reply:
Oh, sensible commit messages, very nice! I guess an interim phase with two systems is fine (famous last words?)